src/contexts/device-size.tsx

```diff
@@ -0,0 +1,97 @@
+import { createContext, useContext, useEffect, useState } from 'react'
+
+const DEFAULT_MOBILE_WIDTH = 728
+const DEFAULT_TABLET_WIDTH = 1000
+
+interface DeviceSize {
+  isDesktop?: boolean
+  isMobile?: boolean
+  isTablet?: boolean
+}
+
+// TODO: does this seem useful as a helper/util?
+const getCSSVariableFromDocument = (
+  variableName: string,
+  options: { asNumber?: boolean } = {}
+): string | number => {
+  const value = getComputedStyle(document.documentElement).getPropertyValue(
+    variableName
+  )
+
+  if (options.asNumber) {
+    return parseInt(value, 10)
+  }
+
+  return value
+}
+
+const DeviceSizeContext = createContext<DeviceSize>(undefined)
+
+const DeviceSizeProvider: React.FC = ({ children }) => {
+  let mobileWidth: number
+  let tabletWidth: number
+  let mobileMediaQueryListObject: MediaQueryList
+  let tabletMediaQueryListObject: MediaQueryList
+
+  if (typeof window !== 'undefined') {
+    mobileWidth =
+      (getCSSVariableFromDocument('--mobile-width-breakpoint', {
+        asNumber: true,
+      }) as number) || DEFAULT_MOBILE_WIDTH
+    tabletWidth =
+      (getCSSVariableFromDocument('--tablet-width-breakpoint', {
+        asNumber: true,
+      }) as number) || DEFAULT_TABLET_WIDTH
+
+    mobileMediaQueryListObject = window.matchMedia(
+      `(max-width: ${mobileWidth}px)`
+    )
+    tabletMediaQueryListObject = window.matchMedia(
+      `(min-width: ${mobileWidth + 1}px) and (max-width: ${tabletWidth}px)`
+    )
+  }
+
+  const getDerivedState = () => {
+    const isMobile = mobileMediaQueryListObject?.matches
+    const isTablet = tabletMediaQueryListObject?.matches
+    const isDesktop = !(isMobile || isTablet)
+    return {
+      isDesktop,
+      isMobile,
+      isTablet,
+    }
+  }
+
+  const [value, setValue] = useState<DeviceSize>(() => getDerivedState())
```

Comment:
Just calling out that if we do use this for hiding components instead of CSS we'll get hydration mismatches, e.g.:
![Screen Shot 2021-12-16 at 10 25 36 AM](https://user-images.githubusercontent.com/1289701/146409985-4d0e4592-b1ad-45a9-8c82-848548714b15.png)

(I just added `if (!isDesktop) return null` to validate the above ☝️, we aren't using it in this way as part of this PR though)

Reply:
This would be perfect to note in a Usage section of a README for this hook! I'm going to make an Asana ticket for documenting the the Context/hook and will link to this comment so we don't lose track of it.